docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,1672 @@
+
+Blobs are small chunks of data. In BlobVM, we divide a file into small blobs and store them as a key-value pair. The key of these small chunks of a file is linked together as the children of the root. This tree is also stored as JSON data against the root's key.
+
+## How to Use a VM
+
+VM modules are served as a binary plugin to the `avalanchego` software. The nodes that want to power their chains using a particular VM, must have the built binary in their `$GOPATH/src/github.com/ava-labs/avalanchego/build/plugins/` (if they haven't built from source), and `$SOURCE_DIR/build/plugins/` if they are building from source. This can also be customized (see [here](../nodes/maintain/avalanchego-config-flags.md#--build-dir-string) for more details). There could be multiple VM plugins in this directory.
+
+## High-Level Overview
+
+Virtual Machines are created as a module, whose binary is registered by a node running `avalanchego`, against the **vmID** (binary file name must be vmID). VMID is a user-defined string that is zero-extended to a 32-byte array and encoded in CB58.
+
+We can build multiple chains with the registered VM. A VM is registered, only when its built binary is present at the required location, as explained [here](#how-to-use-a-vm).
+
+### APIs for a VM
+
+The VM exposes 3 things for interaction:
+
+- **Static and Non-Static Handlers (API)** for making requests to VM methods
+- **Service** for implementing handlers or APIs
+- **Client** for making calls to the service's APIs
+
+### What are Handlers
+
+Handlers serve the response for the incoming HTTP requests. Handlers can also be wrapped with **gRPC** for efficiently making calls from other services such as `avalanchego`. Handler helps in creating APIs. VM implements 2 kinds of handlers:
+
+- **Non-Static Handlers** - They help in interacting with blockchains instantiated by the VM. The API's endpoint will be different for different chains. `/ext/bc/[chainID]`
+- **Static Handlers** - They help in directly accessing VM. These are optional for reasons such as parsing genesis bytes required to instantiate new blockchains. `/ext/vm/[vmID]`
+
+### Registering a VM
+
+While [registering a VM](https://github.com/ava-labs/avalanchego/blob/master/vms/manager.go#L93), we store its **factory** against the **vmID**. As the name suggests, a VM factory can create new VM instances. Using the VM's instance, we can initialize it and can create a new chain. Note that, instantiating a VM is different than initializing it. We initialize the instance of a VM with parameters like genesis, database manager, etc. to create a functional chain. A functional chain doesn't mean the node will start issuing blocks and processing transactions. This will only happen once the node is bootstrapped.
+
+```go
+// Registering a factory inside avalanchego
+func (m *manager) RegisterFactory(vmID ids.ID, factory Factory) error {
+    ...
+    m.factories[vmID] = factory
+    ...
+}
+```
+
+### Instantiating a VM
```

Comment:
Feels like we're duplicating with the `timestampvm` tutorial in a lot of places in the doc [here](https://docs.avax.network/subnets/create-a-virtual-machine-vm#factory). Maybe we should break this + the timestampvm apart to have three docs? 
1. What needs to be implemented for a VM, explains each part of the vm interface and what it does
2. Timestamp tutorial, just references (1) and explains additional stuff we built on top to support timestampvm-specific features
3. Blob tutorial, similarly references (1) and explains additional stuff we built on top to support blobvm-specific features.

cc: @cgcardona @aaronbuchwald

Reply:
Hi @joshua-kim. Thanks for this suggestion. We are moving ahead with this 3-tutorial VM series. I have made a recent commit to support this. Please have a look. Thank you.